src/pip/_internal/req/__init__.py

```diff
@@ -26,6 +28,12 @@ def __repr__(self):
         return f"InstallationResult(name={self.name!r})"
 
 
+_InstallArgs = collections.namedtuple(
+    '_InstallArgs',
+    ['install_options', 'global_options', 'kwargs']
```

Comment:
Why not explode `kwargs` into separate namedtuple attributes and use `install_args._todict()`? I was thinking

```python
_InstallArgs = collections.namedtuple(
    '_InstallArgs',
    'install_options global_options root home prefix warn_script_location use_user_site pycompile',
)

...

install_args = _InstallArgs(
    install_options=install_options,
    global_options=global_options,
    root=root,
    home=home,
    prefix=prefix,
    warn_script_location=warn_script_location,
    use_user_site=use_user_site,
    pycompile=pycompile,
)
```

Reply:
Hmm I can't remember what was the original thought on separating the two (probably had something to do with the original code implantation), anyhow you are absolutely correct, no reason why we should not flatten it all.